pkg/providers/merge/worker.go

```diff
@@ -60,23 +60,56 @@ func (m *merge) startJob(mergeJob *AutoMerge) error {
 		time.Sleep(waitForStatus)
 	}
 
-	commentBody := testCommentBody
-	conmment := &github.IssueComment{
-		Body: &commentBody,
-	}
-	_, _, err = m.opr.Github.Issues.CreateComment(context.Background(),
-		m.owner, m.repo, *pr.Number, conmment)
-	if err != nil {
-		return errors.Wrap(err, "start merge job")
+	canMergeWithoutTests, err := m.canMergeWithoutRunAllTest(mergeJob, pr)
+
+	needRunAllTests := needUpdate || !canMergeWithoutTests
+
+	if needRunAllTests {
+		commentBody := testCommentBody
+		comment := &github.IssueComment{
+			Body: &commentBody,
+		}
+		_, _, err = m.opr.Github.Issues.CreateComment(context.Background(),
+			m.owner, m.repo, *pr.Number, comment)
+		if err != nil {
+			return errors.Wrap(err, "start merge job")
+		}
+		time.Sleep(waitForStatus)
 	}
+
 	mergeJob.Started = true
 	if err := m.saveModel(mergeJob); err != nil {
 		util.Error(errors.Wrap(err, "start merge job"))
 	}
-	time.Sleep(waitForStatus)
 	return nil
 }
 
+func (m *merge) canMergeWithoutRunAllTest(mergeJob *AutoMerge, pr *github.PullRequest) (bool, error) {
+	if mergeJob == nil {
+		return false, errors.New("got a nil merge job")
+	}
+	if pr == nil {
+		return false, errors.New("got a nil pr")
+	}
+
+	if !mergeJob.WithoutTests {
+		return false, nil
```

Comment:
Seems it can not read `without_tests` correctly from database and always failed skip tests here, still debugging.